Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -793,13 +811,23 @@ int __cdecl Dock_Call(unsigned int const &iShip, unsigned int const &iTargetID,
 		// If target not a player in FREIGHTER class ship, ignore request
 		returncode = SKIPPLUGINS;
 
-		uint iType;
-		pub::SpaceObj::GetType(iTargetID, iType);
-		if (!(iType & (OBJ_FREIGHTER | OBJ_TRANSPORT | OBJ_GUNBOAT | OBJ_CRUISER | OBJ_CAPITAL)))
+		uint iTargetType;
+		pub::SpaceObj::GetType(iTargetID, iTargetType);
+		if (!(iTargetType & (OBJ_TRANSPORT | OBJ_GUNBOAT | OBJ_CRUISER | OBJ_CAPITAL)))
 		{
 			return 0;
 		}
 		
+		uint iShipType;
+		pub::SpaceObj::GetType(iShip, iShipType);
+		if (iShipType & (OBJ_TRANSPORT | OBJ_GUNBOAT | OBJ_CRUISER | OBJ_CAPITAL))
+		{
+			PrintUserCmdText(client, L"ERR Your ship is too large for mobile docking!");

```

Comment:
should we be doing PrintUserCmdText/dockPort/response in the bad target type handling code above too?